packages/debug/src/browser/editor/debug-editor-model.ts

```diff
@@ -150,6 +150,26 @@ export class DebugEditorModel implements Disposable {
         return decorations;
     }
 
+    protected async toggleExceptionWidget(): Promise<void> {
+        const { currentFrame } = this.sessions;
+        if (!currentFrame ||
+            !currentFrame.source || currentFrame.source.uri.toString() !== this.uri.toString() ||
```

Comment:
This shouldn't guard against the currentFrame not having source as it's common to have the topmost stack frame point at the exception and not have source:

response.body.stackFrames:

[
    {
        column: 0
        id: 1001
        line: 0
        name: "__cxa_throw"
        source: undefined
    },{
        column: 0
        id: 1002
        line: 24
        name: "main"
        source: Source {name: "main.cpp", path: ... }
    }
]

In this case it returns early and the exception isn't shown:

<img width="533" alt="Screen Shot 2019-09-23 at 11 47 43" src="https://user-images.githubusercontent.com/61341/65420347-d2d03880-ddf8-11e9-8450-6f45fa93aeae.png">

Reply:
Unless the adapter is expected to strip that frame off, which sounds suspicious.